plugins/pulp_rpm/plugins/distributors/yum/metadata/updateinfo.py

```diff
@@ -85,6 +85,38 @@ def _repo_unit_nevra(self, erratum_unit, repo_id):
 
         return repo_unit_nevra
 
+    def _get_package_checksum_tuple(self, package):
+        """
+        Decide which checksum to publish for the given package in the erratum package list.
+
+        Handle two possible ways of specifying the checksum in the erratum package list:
+        - in the `sum` package field as a list of alternating checksum types and values,
+          e.g. ['type1', 'checksum1', 'type2', 'checksum2']
+        - in the `type` and `sums` package fields. It is only the case when the erratum was uploaded
+          via pulp-admin. Only one type of the checksum could be specified this way.
+
+        :param package: package from the erratum package list
+        :type  package: dict
+        :return: checksum type and value to publish
+        :rtype: tuple
+        """
+        package_checksum_tuple = ()
+        repo_checksum_type = self.checksum_type
+        package_checksums = package.get('sum')
+        if package_checksums:
+            checksum_types = package_checksums[::2]
+            checksum_values = package_checksums[1::2]
+            checksums = dict(zip(checksum_types, checksum_values))
```

Comment:
We do not validate this weird alternating checksum list, so the could be anything in the db, so in your case we can't be sure there won't be `IndexError` for example, and in theory `self.checksum_type` could be `None`... 
So more exceptions to catch vs. new data structures?